app/src/main/java/org/mozilla/tv/firefox/utils/Settings.kt

```diff
@@ -51,10 +52,11 @@ class Settings private constructor(context: Context) {
 
     fun shouldShowTVOnboarding(localeManager: LocaleManager, context: Context): Boolean {
         val channelConfig = ChannelConfig.getTvGuideConfig(context)
+        val currentLocale = localeManager.getCurrentLocale(context)
 
         return !preferences.getBoolean(ChannelOnboardingActivity.TV_ONBOARDING_SHOWN_PREF, false) &&
                 channelConfig.isEnabledInCurrentExperiment &&
-                localeManager.currentLanguageIsEnglish(context)
+                currentLocale.languageAndMaybeCountryMatch(channelConfig.enabledInLocales)
```

Comment:
The logic in these two lines (isEnabled && languageAndMaybeCountryMatch) is duplicated in KillswitchLayout.setVisibility. The latter implementation is more complex but I wonder if there's a way we can consolidate them. Can you investigate in the follow-up for the nits?

Reply:
Agreed.  The first fix I wanted to make for this was creating a new class that was experiment and locale aware, that would return a one line `shouldDisplayX` for each feature.  But there were are a few organizational complications there, as well as some refactor work to clarify the difference between that class and `ExperimentProvider`.  We decided to scale back the work, since it felt a little bit aspirational.

Reply:
In that case, it'd be good to add comments to both sections of code mentioning the duplication and where to find it. That way if one changes, there's a larger chance the other will also change.